op-node/rollup/derive/engine_controller.go

```diff
@@ -149,6 +149,7 @@ func (e *EngineController) SetUnsafeHead(r eth.L2BlockRef) {
 	e.metrics.RecordL2Ref("l2_unsafe", r)
 	e.unsafeHead = r
 	e.needFCUCall = true
+	e.rollupCfg.CheckForkActivation(e.log, r)
```

Comment:
We don't always actually activate based on the L2 timestamp, but this is good enough for our logging

Reply:
Out of curiosity, how else is a hardfork activated if not by L2 timestamp?

Reply:
It can be based on L1 block timestamp or the L1 timestamp of the derivation pipeline (technically the timestamp of the highest L1 block which has been processed for derivation)